atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -889,6 +887,29 @@ public String toString() {
         }
     }
 
+    private void truncateInternal(Client client, String tableName) throws TException {
+        for (int tries = 1; tries <= CassandraConstants.MAX_TRUNCATION_ATTEMPTS; tries++) {
+            boolean successful = true;
+            try {
+                client.truncate(internalTableName(tableName));
+            } catch (TException e) {
+                log.error("Cluster was unavailable while we attempted a truncate for table " + tableName + "; we will try " + (CassandraConstants.MAX_TRUNCATION_ATTEMPTS - tries) + " additional time(s). (" + e.getMessage() + ")");
+                if (CassandraConstants.MAX_TRUNCATION_ATTEMPTS - tries == 0) {
+                    throw e;
+                }
+                successful = false;
+                try {
+                    Thread.sleep(new Random().nextInt((1 << (CassandraConstants.MAX_TRUNCATION_ATTEMPTS - tries)) - 1) * 1000);
```

Comment:
Meant it as normal exponential backoff.
The entire case is degenerate and not perf-involved so I didn't think it'd be worth extracting.

```java
	public static void main (String[] args) throws java.lang.Exception
	{
		for (int test = 0; test < 20; test++) {
		int sum = 0;
		System.out.print("[");
		for (int tries=1; tries <= 3; tries++) {
			int iteration = new Random().nextInt((1 << tries)-1)*1000;
			sum += iteration;
			System.out.print(iteration + ",");
		}
		System.out.println("] sum=" + sum);
		}
	}
```

```
[0,2000,6000,] sum=8000
[0,2000,3000,] sum=5000
[0,2000,5000,] sum=7000
[0,1000,2000,] sum=3000
[0,0,0,] sum=0
[0,1000,5000,] sum=6000
[0,2000,6000,] sum=8000
[0,1000,6000,] sum=7000
[0,2000,6000,] sum=8000
[0,2000,1000,] sum=3000
[0,2000,3000,] sum=5000
[0,2000,6000,] sum=8000
[0,2000,1000,] sum=3000
[0,0,6000,] sum=6000
[0,1000,5000,] sum=6000
[0,1000,5000,] sum=6000
[0,0,1000,] sum=1000
[0,1000,1000,] sum=2000
[0,1000,4000,] sum=5000
[0,2000,2000,] sum=4000
```